### PR TITLE
Migrate to Yara-X

### LINKED ISSUE #122: Migrate to Yara-X
It seem as if [Yara-X](https://github.com/VirusTotal/yara-x) is in a stable state now, so I think we should migrate to it

### PR DESCRIPTION
**Pull request by Robin5605**
Closes #122 

Blocked by #121

### CODE REVIEW

**Comment by Robin5605** [comment]
todo: CI

**Comment by sid-maddy** [comment]
> todo: CI

https://github.com/vipyrsec/dragonfly-client-rs/pull/127

File: src/client/models.rs [hunk 1]
```diff
@@ -71,19 +70,20 @@ pub struct RulesResponse {
...
             .values()
             .map(String::as_ref)
             .collect::<Vec<&str>>()
             .join("\n");
 
-        let compiled_rules = Compiler::new()?
-            .add_rules_str(&rules_str)?
-            .compile_rules()?;
+        let mut compiler = yara_x::Compiler::new();
+        compiler.add_source(rules_str.as_str())?;
```
Comment (jonathan-d-zhang) [inline src/client/models.rs:82]: Can't we skip this joining together and just
```rs
for rule_def in self.rules.values() {
    compiler.add_source(rule_def.as_str())?;
}
```

File: src/scanner.rs [hunk 2]
```diff
@@ -250,13 +250,15 @@ pub fn scan_all_distributions(
 /// # Arguments
 /// * `path` - The path corresponding to this file
 /// * `rules` - The compiled rule set to scan this file against
-fn scan_file(file: &mut impl Read, path: &Path, rules: &Rules) -> Result<FileScanResult> {
+fn scan_file(file: &mut impl Read, path: &Path, rules: &yara_x::Rules) -> Result<FileScanResult> {
```
Comment (jonathan-d-zhang) [inline src/scanner.rs:253]: nit
```suggestion
fn scan_file(file: &mut impl Read, path: &Path, rules: &Rules) -> Result<FileScanResult> {
```

**Review by jonathan-d-zhang — CHANGES_REQUESTED** [review]
lgtm except a few comments

**Review by jonathan-d-zhang — APPROVED** [review]
:shipit:

**Pull request merged**